torch/_library/custom_ops.py

```diff
@@ -52,6 +53,11 @@ def custom_op(
             is valid for. If no device type is provided, then the function
             is used as the default implementation for all device types.
             Examples: "cpu", "cuda".
+        manual_schema (None | str): A schema string for the operator. If not
+            provided, we'll infer a schema for the operator from its type
+            annotations. We recommend letting us infer a schema unless you
+            have a specific reason not to (e.g. you're code-generating a schema).
```

Comment:
What do you mean by that?

Reply:
I'm trying to prevent users from using `manual_schema` if type inference works for them.

Reply:
I would put all the details in the docstring here tbh. Not necessarily in this argument but below as its own paragraph.

Also I guess any case where type annotations fail you for any reason is the main use case.
Or a case where our interpretation of the type annotation is not what you want.

Reply:
Added a paragraph in the docstring